oxidation/libparsec/crates/types/src/addr.rs

```diff
@@ -16,6 +16,15 @@ const PARSEC_SCHEME: &str = "parsec";
 const PARSEC_SSL_DEFAULT_PORT: u16 = 443;
 const PARSEC_NO_SSL_DEFAULT_PORT: u16 = 80;
 
+// Url has a special way to parse http/https schemes. This is because those kind
+// of url have special needs (for instance host cannot be empty).
+// The issue is we want our custom parsec scheme to work in a similar fashion, but
+// we cannot just tell Url "apply to parsec:// whatever rules you use for http://".
+// So instead we have to rely on a hack and manually replace our scheme by http before
+// handing it to Url :'(
+// see. https://github.com/servo/rust-url/issues/763
+struct ParsecUrlAsHTTPScheme<'a>(pub &'a Url);
```

Comment:
Use doc-comment

```suggestion
/// Url has a special way to parse http/https schemes. This is because those kind
/// of url have special needs (for instance host cannot be empty).
/// The issue is we want our custom parsec scheme to work in a similar fashion, but
/// we cannot just tell Url "apply to parsec:// whatever rules you use for http://".
/// So instead we have to rely on a hack and manually replace our scheme by http before
/// handing it to Url :'(
/// see. https://github.com/servo/rust-url/issues/763
struct ParsecUrlAsHTTPScheme<'a>(pub &'a Url);
```